pkg/apis/build/v1beta1/buildstrategy_conversion.go

```diff
@@ -136,6 +206,17 @@ func (src *BuildStrategySpec) ConvertFrom(bs v1alpha1.BuildStrategySpec) {
 		})
 	}
 
+	// Add replacement for legacy arguments
+
+	if usesDockerfile {
+		src.Parameters = append(src.Parameters, Parameter{
+			Name:        "dockerfile",
+			Description: "The Dockerfile to be built.",
```

Comment:
ideally we should inline this and https://github.com/shipwright-io/build/pull/1352/files#diff-7cd8b697dab952af9e04af485edc3ac5c177272bbf432f2fb9f13b86ae144b6eR56